test/cppsuite/test_harness/workload/database_operation.cxx

```diff
@@ -208,9 +216,10 @@ database_operation::read_operation(thread_context *tc)
 
         tc->transaction.begin();
         while (tc->transaction.active() && tc->running()) {
-            if (tc->next(cursor) == WT_ROLLBACK)
-                /* We got an error, our transaction has been rolled back. */
-                break;
+            if (tc->next(cursor)) {
+                tc->transaction.rollback();
+                continue;
```

Comment:
Should we sleep here in case `tc->next(cursor)` always fails and requires a rollback ?
Something like :
```
            if(tc->next(cursor)) {
                tc->transaction.rollback();
            } else {
                tc->transaction.add_op();
                tc->transaction.try_rollback();
            }
            tc->sleep();
```